HW19: added Dataclass and replaced usage of the dicts with Dataclass

inovikovaqa911 opened the pull request.

inovikovaqa911 committed:
HW19: added Dataclass and replaced usage of the dicts with Dataclass

ddoroshevskyi commented on conftest.py:
```diff
@@ -8,6 +9,51 @@
 log = logging.getLogger(__name__)
 
 
+# class SensorInfo:
```
I understand that you were following along the lecture but generally it's better to avoid committing commented code.

ddoroshevskyi commented on conftest.py:
```diff
@@ -8,6 +9,51 @@
+            raise ValueError("'model' should not be an empty string")
+        if not isinstance(self.firmware_version, int):
+            raise TypeError("'firmware_version' should be an int")
+        if self.firmware_version <= 0:
```
We'd like to test this for validity from sensor's standpoint. That would be checking that the firmware version is between 10 and 15, both ends included. You could test this with the following code:
```
10 <= self.firmware_version <= 15
```

ddoroshevskyi commented on conftest.py:
```diff
@@ -8,6 +9,51 @@
+            raise ValueError("'firmware_version' should be a positive int")
+        if not isinstance(self.reading_interval, int):
+            raise TypeError("'reading_interval' should be a int")
+        if self.reading_interval <= 0:
```
For reading interval we're interested in checking whether the value here is less than 1.

ddoroshevskyi commented on sensor_test.py:
```diff
@@ -1,33 +1,39 @@
-        set_sensor_reading_interval, reset_sensor_to_factory, update_sensor_firmware, reboot_sensor,
+    get_sensor_info,
+    get_sensor_reading,
+    get_sensor_methods,
```
Only `get_sensor_info` and `get_sensor_reading` are used in this test, so the rest of the fixtures can be safely removed.

inovikovaqa911 committed:
HW19 UPD:
removed redundant code from conftest.py,
updated if conditions in __post_init__ func
removed redundant fixtures in test_sanity func

inovikovaqa911 committed:
HW19 UPD:
removed redundant code from conftest.py,
updated if conditions in __post_init__ func
removed redundant fixtures in test_sanity func
